snuba/cli/cleanup.py

```diff
@@ -55,6 +50,19 @@ def cleanup(
     dataset = get_dataset(dataset_name)
     table = enforce_table_writer(dataset).get_schema().get_local_table_name()
 
-    clickhouse = ClickhousePool(clickhouse_host, clickhouse_port)
-    num_dropped = run_cleanup(clickhouse, database, table, dry_run=dry_run)
-    logger.info("Dropped %s partitions on %s" % (num_dropped, clickhouse_host))
+    if bool(clickhouse_host) ^ bool(clickhouse_port):
+        raise click.ClickException("Provide both Clickhouse host and port or neither")
+
+    # If Clickhouse host and port values are provided, run cleanup on that host,
+    # otherwise run cleanup on each registered cluster.
+    if clickhouse_host and clickhouse_port:
+        clickhouse_connections = [ClickhousePool(clickhouse_host, clickhouse_port)]
+    else:
+        clickhouse_connections = [
+            cluster.get_connection(ClickhouseClientSettings.READWRITE)
+            for cluster in CLUSTERS
+        ]
+
+    for connection in clickhouse_connections:
```

Comment:
I think this will be quite fragile.
We are running this run_cleanup on all clusters and not only on the clusters where the table is present.
This incidentally works now because run_cleanup would pick up the parts to clean by selecting the parts where table = xxx. But this is not guaranteed. 
Passing invalid arguments to a method with the assumption that it will handle them gracefully and generate correct results anyway is dangerous and easy to break.

Any way we can focus only on the cluster that handles the writer for the dataset passed in input ?

Reply:
Yes, thanks this was an oversight

Reply:
I think this is still to broad.
we take all clusters from a dataset. Not only the writer ones. 
Then we run the cleanup for the same table on all of them, while we should be running the cleanup only on the connection for the cluster that holds the table.

Reply:
Ok. Cleanup only makes sense for the writable storages right? Do you know if there's any use case for running cleanup on a table that is not our writable storage table?